api/src/main/java/org/openmrs/module/fhir/api/impl/ConditionServiceImpl.java

```
@@ -47,48 +48,52 @@ public void setDao(FHIRDAO dao) {
 		this.dao = dao;
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.ConditionService#getConditionById(String)
+	 */
 	@Override
-	public Condition getCondition(String id) {
-		org.openmrs.Condition condition = Context.getService(org.openmrs.api.ConditionService.class).getConditionByUuid(id);
-		if (condition == null || condition.isVoided()) {
-			return null;
-		}
-		return FHIRConditionUtil.generateFHIRCondition(condition);
+	public Condition getConditionById(String id) {
+		return ContextUtil.getConditionHelper().getCondition(id);
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.ConditionService#getConditionByObsId(String)
+	 */
+	@Deprecated
 	@Override
 	public Condition getConditionByObsId(String id) {
 		return ConditionStrategyUtil.getConditionStrategy().getConditionById(id);
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.ConditionService#searchConditionById(String)
+	 */
 	@Override
-	public List<Condition> searchConditionById(String id) {
-		org.openmrs.Condition condition = Context.getService(org.openmrs.api.ConditionService.class)
-				.getConditionByUuid(id);
-		List<Condition> conditionList = new ArrayList<Condition>();
-		if (condition != null) {
-			conditionList.add(FHIRConditionUtil.generateFHIRCondition(condition));
-		}
-		return conditionList;
+	public List<Condition> searchConditionById(String id) throws NotImplementedException {
+		return null;
```

I mean throwing NotImplementedException instead of returning null